plugins/es-repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java

```diff
@@ -144,14 +145,11 @@ static AWSCredentialsProvider buildCredentials(Logger logger, S3ClientSettings c
 
     @Override
     public void close() {
-        synchronized (this) {
-            var localClientRef = clientCache;
-            if (localClientRef != null) {
-                localClientRef.decRef();
-            }
-            clientCache = null;
-            clientSettingsCache = null;
+        // the clients will shutdown when they will not be used anymore
```

Comment:
It's edge case, but if a thread tries to access the cache, while the service is closing down, then we can run into issues, so I'd keep the synchronized also here.

Reply:
Good point, will do.